src/fparser/two/tests/test_c99preprocessor.py

```diff
@@ -0,0 +1,569 @@
+    '''Test that various forms of #if, #ifdef, #ifndef are recognized'''
+    result = Cpp_If_Stmt(line)
+    assert str(result) == ref
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line', [
+    None, '', ' ', '#ifdfe', '#if', '#ifdef', '#ifdef two macros'])
+def test_incorrect_if_stmt(line):
+    '''Test that incorrectly formed #if statements raise exception'''
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = Cpp_If_Stmt(line)
+    assert "Cpp_If_Stmt: '{0}'".format(line) in str(excinfo.value)
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line', ['#elif CONDITION', '  #  elif   CONDITION '])
+def test_elif_stmt(line):
+    '''Test that #elif is correctly recognized'''
+    ref = '#elif CONDITION'
+    result = Cpp_Elif_Stmt(line)
+    assert str(result) == ref
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line', [None, '', ' ', '#elfi', '#elif'])
+def test_incorrect_elif_stmt(line):
+    '''Test that incorrectly formed #elif statements raise exception'''
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = Cpp_Elif_Stmt(line)
+    assert "Cpp_Elif_Stmt: '{0}'".format(line) in str(excinfo.value)
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line', ['#else', '  # else  '])
+def test_else_stmt(line):
+    '''Test that #else is correctly recognized'''
+    result = Cpp_Else_Stmt(line)
+    assert str(result) == line
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line', [None, '', ' ', '#esle', '#elseA', '#Aelse'])
+def test_incorrect_else_stmt(line):
+    '''Test that incorrectly formed #else statements raise exception'''
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = Cpp_Else_Stmt(line)
+    assert "Cpp_Else_Stmt: '{0}'".format(line) in str(excinfo.value)
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line', ['#endif', '  #  endif  '])
+def test_endif_stmt(line):
+    '''Test that #endif is correctly recognized'''
+    result = Cpp_Endif_Stmt(line)
+    assert str(result) == line
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line',
+                         [None, '', ' ', '#ednif', '#endifA', '#Aendif'])
+def test_incorrect_endif_stmt(line):
+    '''Test that incorrectly formed #endif statements raise exception'''
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = Cpp_Endif_Stmt(line)
+    assert "Cpp_Endif_Stmt: '{0}'".format(line) in str(excinfo.value)
+
+
+def test_parse_define_outside_subroutine():
+    f2003_parser = ParserFactory().create(std='f2003')
+
+    code = '#define MACRO\nSUBROUTINE FOO\ncall sub()\nEND SUBROUTINE FOO\n'
+    ref = '#define MACRO\nSUBROUTINE FOO\n  CALL sub\nEND SUBROUTINE FOO'
+    reader = get_reader(code)
+    result = f2003_parser(reader)
+    assert str(result) == ref
+
+
+def test_parse_empty_ifdef():
+    f2003_parser = ParserFactory().create(std='f2003')
+
+    code = '''
```

Comment:
For this test and subsequent tests could you reduce the ammount of code by making the input the same as the output, or the input a stripped version of the output if you prefer, rather than replicating the code?